sound/soc/sof/intel/hda-loader.c

```diff
@@ -246,6 +247,7 @@ static int cl_cleanup(struct snd_sof_dev *sdev, struct snd_dma_buffer *dmab,
 	hstream->bufsize = 0;
 	hstream->format_val = 0;
 
+	release_firmware(plat_data->fw);
```

Comment:
I see a release_firmware in sof_pci_remove()? Are you positive this was not removed based on experiments or is this based on code analysis only?

It looks like this entire firmware management is broken, the request_firmware() is called in the .load_firmware callback but the corresponding .unload_firmware() is not used, and all devices do this on their own. That's weird. @lgirdwood should we fix this?

I also don't think we should have a snd_sof_firmware_memcpy(), this is completely Intel-specific and does not belong in the generic loader.c

Reply:
@plbossart Sorry, it is only based on code analysis. And yes, it will be released in sof_pci_remove().
But it seems to be odd. We call snd_sof_load_firmware() in sof_resume(), but release_firmware() in sof_pci_remove(). It means it is possible that we request firmware many times but only release the latest one.